src/include/storage/store/table.h

```diff
@@ -49,7 +62,13 @@ struct TableScanState {
     virtual ~TableScanState() = default;
     DELETE_COPY_DEFAULT_MOVE(TableScanState);
 
-    virtual void initState(transaction::Transaction* transaction, NodeGroup* nodeGroup) {
+    virtual void setToTable(Table* table_, std::vector<common::column_id_t> columnIDs_,
```

Comment:
Up to u but seeing rel table specific field `direction & resetCachedBoundNodeSelVec` is a bit weird.

Reply:
Yeah it's a bit annoying, but I will see if I can improve this in the coming refactoring.